examples/ex4p.cpp

```diff
@@ -269,7 +290,7 @@ int main(int argc, char *argv[])
       sol_sock << "solution\n" << *pmesh << x << flush;
    }
 
-   // 17. Free the used memory.
+   // 18. Free the used memory.
    delete pcg;
    delete prec;
    delete hfes;
```

Comment:
There are some unnecessary comments in ex4 lines: 272, 289 and ex4p lines: 319, 336 that I think they should be removed. There are not introduced in this PR though

Reply:
Right, these are already in master, not from this PR. @v-dobrev Should we keep these commented definitions of z in ex4 and ex4p?

Reply:
I think these are left there for the reader (users) in case they want to try another solution that depends on `z`. Do you find them more confusing than helpful? 😄

Reply:
@v-dobrev Thanks for clarifying. This may be the first time I've seen commented code in an example. It is helpful and a little confusing, but it might be more helpful it were less confusing. Maybe we should add a comment saying something like "uncomment if f depends on z". I am fine with leaving these lines as they are, but a comment might be a good idea.

Reply:
I agree, adding a comment as you suggest will be an improvement.

Reply:
Comments are added.